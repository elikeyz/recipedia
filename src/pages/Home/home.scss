.landing-background {
  background-image: url('../../static/images/background.jpg');
  height: 100%;
  background-size: cover;
}

.overlay {
  height: 100%;
  background-color: rgba(255,255,255,0.6);
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.landing-container {
  text-align: center;

  button {
    background-color: #FF4A83;
    border: 1px solid #FF4A83;
    width: 25%;

    &:hover {
      background-color: white;
      color: #FF4A83;
      border: 1px solid #FF4A83;
    }
  }
}

@media only screen and (max-width: 576px) {
  .landing-container {
    button {
      width: 50%;
    }
  }
}
