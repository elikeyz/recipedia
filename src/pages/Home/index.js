/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { Container, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import './home.scss';

class Home extends Component {
  render() {
    return (
      <main>
        <div className="overlay">
          <Container>
            <h1>Welcome to Recipedia</h1>
            <h2>Explore thousands of recipes from across the globe</h2>
            <br />
            <Link to="/recipes">
              <Button size="lg">Explore</Button>
            </Link>
          </Container>
        </div>
      </main>
    );
  }
}

export default Home;
