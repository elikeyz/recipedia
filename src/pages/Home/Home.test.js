import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Home Component', () => {
  test('renders', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
