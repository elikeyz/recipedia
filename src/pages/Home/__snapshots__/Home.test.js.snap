// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders 1`] = `
<main
  className="landing-background"
>
  <div
    className="overlay"
  >
    <Container
      className="landing-container"
      fluid={false}
    >
      <h1>
        Welcome to Recipedia
      </h1>
      <h2>
        Explore thousands of recipes from across the globe
      </h2>
      <br />
      <Link
        to="/recipes"
      >
        <Button
          active={false}
          disabled={false}
          size="lg"
          type="button"
          variant="dark"
        >
          Explore
        </Button>
      </Link>
    </Container>
  </div>
</main>
`;
